### DISCUSSION TITLE
v0.1.0

### DISCUSSION THREAD

**Discussion by philiplinden** [philiplinden/spacetime#20] [Announcements]
This is a checkpoint of work done in late 2023 and early 2024. It is the culmination of much experimentation. 

The tag does **not** indicate a _working_ release---it certainly is not one. It is a pinned "save point" before continuing to change everything.

## What's Changed
* simple mesa+nyx example by @philiplinden in https://github.com/philiplinden/spacetime/pull/1
* Prototype Rust implementation using Bevy by @philiplinden in https://github.com/philiplinden/spacetime/pull/3
* Model touch ups for consistency by @philiplinden in https://github.com/philiplinden/spacetime/pull/7
* Rust implementation of Marco Polo demo by @philiplinden in https://github.com/philiplinden/spacetime/pull/8
* Change workflow to make rust docs by @philiplinden in https://github.com/philiplinden/spacetime/pull/9
* Time dilation experiments by @philiplinden in https://github.com/philiplinden/spacetime/pull/10
* Rust docs by @philiplinden in https://github.com/philiplinden/spacetime/pull/11
* Time dilation by example by @philiplinden in https://github.com/philiplinden/spacetime/pull/12
* Feeling out straw-man project architectures with Bevy by @philiplinden in https://github.com/philiplinden/spacetime/pull/13
* pocket solar system by @philiplinden in https://github.com/philiplinden/spacetime/pull/14
* Site and docs overhaul by @philiplinden in https://github.com/philiplinden/spacetime/pull/15
* Rebrand clocss-abm -> spacetime by @philiplinden in https://github.com/philiplinden/spacetime/pull/16
* More technical documentation by @philiplinden in https://github.com/philiplinden/spacetime/pull/17
* CHECKPOINT: accumulated dev updates by @philiplinden in https://github.com/philiplinden/spacetime/pull/18
* 🤷‍♂️ by @philiplinden in https://github.com/philiplinden/spacetime/pull/19

## New Contributors
* @philiplinden made their first contribution in https://github.com/philiplinden/spacetime/pull/1

**Full Changelog**: https://github.com/philiplinden/spacetime/commits/v0.1.0

<hr /><em>This discussion was created from the release <a href='https://github.com/philiplinden/spacetime/releases/tag/v0.1.0'>v0.1.0</a>.</em>

**Comment by ChristopherRabotin** [comment]
> [nyx_space](https://nyxspace.com/). The documentation is... sparse. Expect to spend a lot of time in the source code or DMing the author. There is a wide set of tools tailored specifically for LEO missions up to lunar trajectories, including thrust maneuvers and simulated orbit determination from terrestrial ground stations.

Hahaha this is generous! I promise I'm working on this as part of the official version [2.0.0](https://github.com/nyx-space/nyx/milestone/7). I've started writing [some examples](https://github.com/nyx-space/nyx/tree/master/examples) for Nyx, and the ANISE [documentation](https://nyxspace.com/anise/) is a notch better. But overall, I strongly agree that I need to work a bunch on the documentation, and in the meantime, absolutely do not hesitate to reach out!

I really like that you've set up a documentation website already. What tool did you use for the [arch documentation](https://philiplinden.github.io/spacetime/architecture.html), it looks pretty nice!

**Reply by philiplinden** [reply]
The website uses
- [MkDocs](https://www.mkdocs.org/) to build a website from markdown, with [plugins](https://github.com/mkdocs/catalog). Config in `mkdocs.yml`, build example in the github actions workflows.
  - `mkdocs-material` https://github.com/squidfunk/mkdocs-material
  - `include_dir_to_nav` https://github.com/mysiki/mkdocs_include_dir_to_nav
  - `git-authors` https://github.com/timvink/mkdocs-git-authors-plugin
- [nbconvert](https://github.com/jupyter/nbconvert) is used on the Jupyter notebooks to convert them into static webpages. This used to run in the workflow too, but I took it out because it was annoying to maintain.
- [draw.io](https://www.drawio.com/) for flow charts and block diagrams. These are saved as `.xml` but I export them to images and embed those on the website. I used the "sketch" stylization feature.

This is my go-to setup lately. I've been using it for a while and I love it.
